app/models/ai_persona.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+class AiPersona < ActiveRecord::Base
+  class MultisiteHash
+    def initialize(id)
+      @hash = Hash.new({})
+      @id = id
+
+      MessageBus.subscribe(channel_name) { |message| @hash[message.data] = {} }
```

Comment:
What is the need behind messagebus?

Reply:
we need to communicate cross process to notify everyone the in memory cache is out of date. 

distributedcache can be used here ... but we would need a "version" to cache and finding that is hard. this construct is easier to reason about.

Reply:
OK cool makes sense :+1: